script.js

```diff
@@ -9,13 +9,42 @@ const small_film_set = [
     { id: 6, title: "12 Angry Men", year: 1957, votes: 164558, rating: 8.9, rank: 6, category: "Western" }
 ]
 
+function saveItem() {
+    const validationResult = $$("form").validate();
+    if (validationResult) {
+        const item_data = $$("form").getValues();
+        $$("table").add(item_data);
+        $$("form").clear();
+        webix.message({
+            text: `Validation is successful`,
+            type: "success",
+            expire: 1000
+        });
+    }
+}
+
+function clearValidation() {
+    webix.confirm(`Do you realy want to clear validation?`)
```

Comment:
(1) why chaining promises for sync code? `.clearValidation` and `.clear` should be placed into one and the same handler.

https://medium.com/@fagnerbrack/promises-sync-code-disaster-e9d41a3c7279

"Besides the obvious performance implications of abusing the Promises chain by running synchronous functions unnecessarily, you are also breaking the reader’s expectation by changing the intent of how the code is supposed to work."

(2) and minor issue - end users do not know anything about technical details of the app, so the message text should ask them about clearing the form itself, not validation.

Reply:
It’s very clear article, thank you. Already fixed, including text message.